TTC.Deployment.AmazonWebServices/Deployer.cs

```diff
@@ -22,28 +23,37 @@ public class Deployer
         private readonly AmazonCloudFormationClient _cloudFormationClient;
         private readonly AmazonS3Client _s3Client;
         private readonly AmazonIdentityManagementServiceClient _iamClient;
-        private readonly AwsConfiguration _awsConfiguration;
         private readonly AmazonAutoScalingClient _autoScalingClient;
-        private readonly AmazonSecurityTokenServiceClient _securityTokenServiceClient;
 
-        public Deployer(AwsConfiguration awsConfiguration) {
-            _awsConfiguration = awsConfiguration;
-            _securityTokenServiceClient = new AmazonSecurityTokenServiceClient(_awsConfiguration.AwsEndpoint);
+        private readonly AwsConfiguration _awsConfiguration;
 
+        public Deployer(AwsConfiguration awsConfiguration) {
             // TODO: don't use EnvironmentAWSCredentials, because it doesn't support environment variables!
             var credentials = awsConfiguration.Credentials ?? new EnvironmentAWSCredentials();
             
-            if (!string.IsNullOrEmpty(_awsConfiguration.RoleName))
+            if (awsConfiguration.AssumedRole != null)   
             {
-//                var clientId = Guid.NewGuid();
-//                const string sessionName = "Net2User";
+                var securityTokenServiceClient = new AmazonSecurityTokenServiceClient(awsConfiguration.AwsEndpoint);
+
+                var assumeRoleResult = securityTokenServiceClient.AssumeRole(new AssumeRoleRequest
+                {
+                    RoleArn = awsConfiguration.AssumedRole.Arn,
+                    DurationSeconds = 3600,
+                    RoleSessionName = "Net2User",

```

Comment:
So, when you assume a role you give that session a name.  It should probably be related to the deployment. perhaps {stack_name}_deployment_{datetime.now} or something?

`Net2User` is clearly the result of some flailing about when trying to create a new assume role session when one already exists with no name or NetUser or something.